File: action-type/16-state/java/EndState.java
package main;

// 结束状态
public class EndState implements LeaveRequestState {
    
    @Override
    public void submit(LeaveRequestContext ctx) {
    
        System.out.println("[结束] 流程已结束，不能再提交");
    }
    
    @Override
    public void approve(LeaveRequestContext ctx) {
    
        System.out.println("[结束] 流程已结束，不能再审批");
    }
    
    @Override
    public void reject(LeaveRequestContext ctx) {
    
        System.out.println("[结束] 流程已结束，不能再拒绝");
    }
    
    @Override
    public void cancel(LeaveRequestContext ctx) {
    
        System.out.println("[结束] 流程已结束，不能再撤销");
    }
    
}

File: action-type/16-state/go/end_state.go
package main

import "fmt"

// EndState 结束状态

type EndState struct{}

func (s *EndState) Submit(ctx *Context)  { fmt.Println("[结束] 流程已结束，不能再提交") }
func (s *EndState) Approve(ctx *Context) { fmt.Println("[结束] 流程已结束，不能再审批") }
func (s *EndState) Reject(ctx *Context)  { fmt.Println("[结束] 流程已结束，不能再拒绝") }
func (s *EndState) Cancel(ctx *Context)  { fmt.Println("[结束] 流程已结束，不能再撤销") }
